src/ompparser.yy

```diff
@@ -2318,9 +2318,36 @@ mapper_identifier_optseq : type_var
 mapper_identifier : IDENTIFIER_DEFAULT { ((OpenMPDeclareMapperDirective*)current_directive)->setIdentifier(OMPD_DECLARE_MAPPER_IDENTIFIER_default);}
                   | EXPR_STRING { ((OpenMPDeclareMapperDirective*)current_directive)->setIdentifier(OMPD_DECLARE_MAPPER_IDENTIFIER_user); ((OpenMPDeclareMapperDirective*)current_directive)->setUserDefinedIdentifier($1); }
                   ;
-
-type_var : EXPR_STRING { ((OpenMPDeclareMapperDirective*)current_directive)->setTypeVar($1); }
+         
+type_var : EXPR_STRING { 
+               if (user_set_lang == Lang_C || auto_lang == Lang_C) { 
+                   const char * _type_var = $1;
+                   std::string type_var = (std::string)_type_var;
```

Comment:
To avoid undefined behavior, please use standard std::string constructor and method for const char* conversion.
e.g. std:string(const char*) to get a string.
some_string_x.c_str() to get a const char*.
Search the official manual for reference, which is more reliable.

Also, please always check the code formatting before pushing, such as space, indent, etc.